### PR TITLE
fix: update kms service endpoint

### PR DESCRIPTION
**Pull request by MatthewLemmond**
### Description
add check for service endpoint for kms and only allow values 'private' and 'public-and-private'

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

The KMS endpoints when creating a resource instance (`use_data` and `use_hs_crypto` are false and `name` is provided in the `var.key_management` block) will use `private` or `public-and-private` as inputs via `var.key_management.service_endpoints`.

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

**Comment by MatthewLemmond** [comment]
Added a new service endpoints field to the key management variable and passed it through to the kms submodule, allowed `private` and `public-and-private` to keep the parity with the other endpoints.

File: kms/variables.tf [hunk 1]
```diff
@@ -83,14 +84,14 @@ variable "keys" {
...
-  validation {
-    error_message = "Service endpoints can only be `public`, `private`, or `public-and-private`."
-    condition     = contains(["public", "private", "public-and-private"], var.service_endpoints)
-  }
-}
+#   validation {
+#     error_message = "Service endpoints can only be `public`, `private`, or `public-and-private`."
+#     condition     = contains(["public", "private", "public-and-private"], var.service_endpoints)
+#   }
+# }
```
Comment (ocofaigh) [inline kms/variables.tf:96]: @MatthewLemmond can you remove the code comments?

**Comment by MatthewLemmond** [comment]
Updated to remove the comment block, added to the description on the root level service endpoints variable, and added the key_management.service_endpoints to all override.json files.

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**